### ISSUE TITLE
Support for Princess Smart Dehumidifier 

### ISSUE THREAD

**Issue by jamesjharris** [dennis1804/homewizard-climate-hass#7]
Hi there, I have the Princess Smart Dehumidifier working in my HomeWizard Climate app. Does this need a separate integration to get this to work. Forgive me, I'm pretty new to HASS so any help is really appreciated :) 

James

**Comment by dennis1804** [comment]
Hi James, it will need some sort of tweaking to get the dehumidifier to work, you could install this integration and see if the device is found but most options probably won't work yet. 
maybe it will even be disabled, but when you give me the name of the device i can add it to the integration and get some basic functionality working.

after that we can take a look at the options the device provides and maybe add it to this codebase.

Kind regards,
Dennis

**Comment by PietPiraat80** [comment]
Hello Dennis,

I have the Princess 01.353120.02.001. this is probably the same model as the original poster because there is only one. It would be great if you could add it to your integration. If I can do anything to help please ask.

Regards,
Sebastiaan

**Comment by dennis1804** [comment]
Hi,

Can you try to install this plugin and see if it finds a disabled entity, once i have the name of the identity i can enable it in this plugin.
It might need some follow up, see what kind of playloads it supports etc. But i first need the entity name :)

Kind regards,

**Comment by dennis1804** [comment]
thanks for that, so the name is dehumidifier, i added it to the other repo (https://github.com/dennis1804/homewizard-climate-websocket) so now it should find the device when you get the latest update.
if you want you can checkout that repo and run the ./debug.sh which is in the root of the project. in there you can set the username / password for homewizard and it tries to find all the devices including the states which we can trigger.
that makes it possible to do more things like adjust the fanspeed, swing mode etc.

**Comment by PietPiraat80** [comment]
It works, my dehumidifiers are added. As you predicted most of the functions are missing. I would love to help further but i have no idea how to run the ./debug.sh file. I cant find the file on my Home Assistant installation. So I downloaded it from the site but I can't run it on my windows pc. Could you please give me some directions on how to do this? Thanks again.

**Comment by PietPiraat80** [comment]
Ok, I figured it out and got the file running. The results are in the attached file. [log.txt](https://github.com/dennis1804/homewizard-climate-hass/files/11505143/log.txt) During the process I pushed all the buttons in the app hoping to get better results. But honestly, I have no idea what I'm doing. So if you need my help isolating the various commands don’t hesitate to ask.

**Comment by dennis1804** [comment]
Thanks, thats really useful! I'll try and add the right payloads for your device

**Comment by dennis1804** [comment]
hi @PietPiraat80, i did some changes to the backend (https://github.com/dennis1804/homewizard-climate-websocket) and it should now be able to test the different states which can be set on the dehumidifier. 
can you please verify that by running the new changes (pull or download the new zip) and running the same script (debug.sh)

it should now set your dehumidifier to:

mode: dehumidify
target_humidity: 50
fan_speed: low
swing: True

if everything works correctly i can add it to this codebase so you can also control it with the homeassistant app :)

thanks,

**Comment by PietPiraat80** [comment]
I did what you asked and it worked. I did have to turn on the dehumidifier manually first though. here is the log 
[log_2.txt](https://github.com/dennis1804/homewizard-climate-hass/files/11513749/log_2.txt).

**Comment by PietPiraat80** [comment]
Now I just need a card to control them in HASS. I have found multiple humidifier cards that actually do the same as a dehumidifier . Thing is, they are looking for humidifiers and the enities that are created by the plugin are climate devices. Is it possible to change this in the plugin?

**Comment by dennis1804** [comment]
Yes thats the thing i still need to implement in this project. But i first wanted to validate that the functions worked :)

**Comment by PietPiraat80** [comment]
As mentioned above, it works great 👍 No need to rush things.

**Comment by jamesjharris** [comment]
Hi @dennis1804 , sorry for my radio silence. I've also managed to get this working on my end, I have also been able to get the basic functions working and visible in NodeRed for basic on/off flows. Looking forward to also helping test the next step when you implement into the project. Many thanks for your help :) 

Kind regards,
James

**Comment by dennis1804** [comment]
hi Sorry for the delay, had some issues getting my development environment up and running. i added some code to the feature/add-dehumidifier (#9) branch.
could you check that one out and see if the features work? kinda hard to debug without the dehumidifier :)

Kind regards,

**Comment by PietPiraat80** [comment]
As I have mentioned I am a complete noob. But this is what I tried; I tried installing the websocket using SSH, Then I restarted HomeAssistant. I also reinstalled the integration, but nothing changed. I must be doing something wrong but I have no idea what. Any help is greatly appreciated.

**Comment by dennis1804** [comment]
Websocket update is already included in that branch so you won't need to install that manually.

Just download [this](https://github.com/dennis1804/homewizard-climate-hass/archive/refs/heads/feature/add-dehumidifier.zip) zip and replace the custom component that is currently running on your home assistant instance.
You can replace it with ftp to your home assistant server or ssh.

Folder should look like this:
'config/custom_components/homewizard_climate/climate.py' (and of course the other files that are in the zip)
After you replaced the folders a restart of home assistant should load the new component.

**Comment by BrendanxP** [comment]
You are an absolute legend for getting the dehumidifier (mostly) working!!
I am willing to help out development as much as I can, and as much as you have time to work on this.
I am currently running the lastest version of your repo and for starters I can summarize current and expected behaviour.

Current behavior
1. It has a temperature input setting, no clue what it controls, I don't think it does anything. It does receive and show the same temperature reading as the Climate app does.
2. It has operation toggle for: heat, cold, off. Heat automatically changes to cold when selected, and cold means on. Off means off.
3. It has a fan mode toggle: off, on, low, medium, high. Low, medium and high all redirect to on, which (also) means the machine is on. Off will toggle the machine off, just like the operation toggle. It is not able to in fact switch between the low and high fan mode of the unit, it is simply another on/off switch.
4. Lastly, the toggle swing mode. It has the options off and horizontal. Horizontal redirects to off, thus the toggle is completely useless in it's current state.

Now to summarize the actual functionality in the home wizard climate app:
- it shows the temperature but does not act on it
- it shows the current humidity
- it has a slider to set the desired humidity level
- it has a timer function (this can have least priority as HA can be the "timer")
- it has an on off toggle
- It has a mode toggle: "Dehumidify", "Automatic", "Continuous", "Laundry" and "Fan".
- it has a fan toggle: High and low
- it has a swing toggle: on and off
- it has a child lock: locked and unlocked

The integration in the Google Home app has all the same features except for:
- it lacks the fan speed toggle (simply not present)
- the mode toggles are identical but lower case

If there is anything I can do to help please let me know. Especially after the 17th of August I will have plenty of time to test/troubleshoot/send logs if you please.

**Comment by Pascal4Games** [comment]
Hey,
If you are still interested:

https://github.com/Pascal4Games/homewizard-climate-hass

Works for the Dehumidifier.
It shows the temperature (after change in value)
It shows the humidity (after change in value)
You can control the humidity level
You can toggle it on (first time make sure the device is off)
You can change the fashion
You can change the fan speed (first time make sure the device is low)
You can change the swing (first time make sure the device is off)

It's not yet possible
Timer and child lock.

I haven't figured out how to add these two options yet.

If there are any problems, please let me know.

![afbeelding](https://github.com/dennis1804/homewizard-climate-hass/assets/62812321/cc237007-ffc2-484d-b862-33d1e89b7b29)
